Library/Homebrew/test/support/helper/cmd/brew-verify-undefined.rb

```diff
@@ -0,0 +1,40 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "cli/parser"
+
+UNDEFINED_CONSTANTS = %w[
+  Cask::Cask
+  Formula
+  Formulary
+  Homebrew::API
+  Tap
```

Comment:
Nice work @apainintheneck! Any other big constants you think we could add in here? I'm happy to take a look and suggest some if that's be helpful. My thinking would just be to add various `require` to `global.rb` and use `brew prof` to see which ones have the biggest impact on a `brew ruby -e 'puts :foo'` (or similar quick command that goes through Ruby-land) output.

Reply:
There are a few more that we don't want to be loaded at start up but they are required by some of the existing files in this list so it's not strictly necessary to add them here. I'm thinking of `FormulaInstaller` and `Keg` in this case. That being said the check here is really simple so there is essentially no cost to us expanding this list. Feel free to take a look and add any that seem mildly suspicious to the list. `Homebrew::API` is an example of a borderline one that I decided to add on a whim.

Reply:
Thanks @apainintheneck. Inspired by your great work here: I've opened https://github.com/Homebrew/brew/pull/18065 as a follow-up.

I feel pretty confident after this that I can stop asking people to not add them to `global.rb` as CI will now catch this instead. Couldn't have done it without your idea here, wonderful stuff 👏🏻 😍